### PR TITLE
Update how default auth policy is set in templates

### PR DESCRIPTION
**Pull request by captainsafia**
Address https://github.com/dotnet/aspnetcore/issues/19580

### CODE REVIEW

File: src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/Startup.cs [hunk 1]
```diff
@@ -116,12 +116,10 @@ public void ConfigureServices(IServiceCollection services)
 
 #endif
 #if (OrganizationalAuth)
-            services.AddControllersWithViews(options =>
```
Comment (pranavkm) [inline src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/Startup.cs:119]: You'll need to keep the call to `AddControllersWithViews`

File: src/ProjectTemplates/Web.ProjectTemplates/content/StarterWeb-CSharp/Startup.cs [hunk 2]
```diff
@@ -109,12 +109,9 @@ public void ConfigureServices(IServiceCollection services)
 #endif
 #if (OrganizationalAuth)
 
-            services.AddControllersWithViews(options =>
```
Comment (pranavkm) [inline src/ProjectTemplates/Web.ProjectTemplates/content/StarterWeb-CSharp/Startup.cs:112]: Same as above

File: src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/Startup.cs [hunk 3]
```diff
@@ -116,12 +116,10 @@ public void ConfigureServices(IServiceCollection services)
...
+
+            services.AddAuthorization(options =>
             {
-                var policy = new AuthorizationPolicyBuilder()
-                    .RequireAuthenticatedUser()
-                    .Build();
-                options.Filters.Add(new AuthorizeFilter(policy));
+                options.FallbackPolicy = options.DefaultPolicy;
```
Comment (pranavkm) [inline src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/Startup.cs:122]: Could we include the snippet that explains what this does:

> // By default, all incoming requests will be authorized according to the default policy

It's not common to include comments in code, but fallback policy is a little new and has no documentation. So putting it here might help a bit

Comment (captainsafia) [reply src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/Startup.cs:130]: I added the comment.

I'm also planning on adding some docs about this to as recommended in the original issue.

File: src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/Startup.cs [hunk 4]
```diff
@@ -124,6 +124,12 @@ public void ConfigureServices(IServiceCollection services)
                 options.Filters.Add(new AuthorizeFilter(policy));
```
Comment (pranavkm) [inline src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/Startup.cs:124]: You would want to remove this. Basically we're replacing the filter with an auth policy, but you still want to keep MVC around.

**Review by pranavkm — COMMENTED** [review]
Would you mind getting a sign off from @javiercn \ @SteveSandersonMS? I wasn't in the meeting they landed on this solution, so it would be good to get their signoff.

**Review by SteveSandersonMS — APPROVED** [review]
Looks correct to me! Thanks @captainsafia.

**Pull request merged**